packages/flutter_tools/lib/src/build_system/targets/shader_compiler.dart

```diff
@@ -17,18 +17,47 @@ import '../../base/logger.dart';
 import '../../build_info.dart';
 import '../../convert.dart';
 import '../../devfs.dart';
-import '../../device.dart';
 import '../build_system.dart';
 
 /// The output shader format that should be used by the [ShaderCompiler].
 enum ShaderTarget {
-  impellerAndroid(<String>['--runtime-stage-gles', '--runtime-stage-vulkan']),
-  impelleriOS(<String>['--runtime-stage-metal']),
-  impellerSwiftShader(<String>['--runtime-stage-vulkan']),
-  sksl(<String>['--sksl']);
+  android(<String>['--runtime-stage-gles', '--runtime-stage-vulkan']),
```

Comment:
Could this whole enum just be a function that maps `TargetPlatform` to `List<String> stages`?

Reply:
which would then simplify all the callsites where we have to pass around both `ShaderTarget` AND `TargetPlatform`

Reply:
Yeah, I think it should be safe to always include the SkSL target with any impeller target. At least for now.

Reply:
It could. I'll do that.